### PR TITLE
[Census TMCF] Parsing (enhanced) TMCF and CSV for Census tables.

### PR DESCRIPTION
**Pull request by jehangiramjad**
In this PR, we have the following:

1. Parsing opaque and simple variable mappings in the Census (enhanced) TMCF format + CSV
2. Produce the processed traditional TMCF and corresponding CSV files with the opaque mappings expanded + geoIDs resolved
3. Unit tests

In the next PR, this will be wrapped up in a processor which can retrieve lots of input CSV files from a POST request and produce the parsed output for each.

The output produced from the test cases was validated manually against the import tool which produces no errors.

### CODE REVIEW

File: scripts/us_census/enhanced_tmcf/process_etmcf.py [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+    csv_out = os.path.join(output_folder, output_csv_filename + ".csv")
+
+    # Process the enhanced TMCF to identify the new columns to be introduced in the csv.
+    with open(etmcf_filepath, 'r') as etmcf, open(tmcf_out, 'w') as out_tmcf:
+        lines = etmcf.readlines()
+
+        output_started = False
+        for line in lines:
+            out_line = line
+            if line.startswith("Namespace:"):
```
Comment (beets) [inline scripts/us_census/enhanced_tmcf/process_etmcf.py:243]: not sure if we need to be stricter with ensuring `Namespace` is immediately before `T=` line, but this is good for now. add a TODO?

**Review by beets — APPROVED** [review]
this is nice!

in general, not sure how strict we need to be with parsing - we currently assume single spaces, no trailing whitespaces, but we might need to (sprinkle `.trim()` etc.

File: scripts/us_census/enhanced_tmcf/testdata/input/ECNBASIC2012.EC1200A1.tmcf [hunk 2]
```diff
@@ -0,0 +1,54 @@
+Namespace:
+T=http://data.census.gov/table/EC1200A1
```
[the remark this answers is not in the document]
Comment (jehangiramjad) [reply scripts/us_census/enhanced_tmcf/testdata/input/ECNBASIC2012.EC1200A1.tmcf:2]: For now, I am replacing the "https://" from all strings. FYI

File: scripts/us_census/enhanced_tmcf/process_etmcf.py [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+    # format example: data.census.gov/table/EC1200A1:ESTAB?naics2012={0}&taxstat={1}&typop={2}
+    format: str
+
+    # ordered_csv_column_names example: ["NAICS2012", "TAXSTAT", "TYPOP"]
+    # The order should correspond to the {0}, {1}, {2}, ... in the format string above.
+    ordered_csv_column_names: List[str]
+
+
+def _replace_ce_t_with_filename(line: str, csv_out_name: str) -> str:
+    return line.replace(":T", ":" + csv_out_name)
```
[the remark this answers is not in the document]
Comment (jehangiramjad) [reply scripts/us_census/enhanced_tmcf/process_etmcf.py]: Done, Thanks!

File: scripts/us_census/enhanced_tmcf/process_etmcf.py [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+    with open(input_csv_filepath, 'r') as input_csv:
+        csv_reader = csv.DictReader(input_csv)
+        for row_dict in csv_reader:
+            if GEO_ID_COLUMN not in row_dict.keys():
+                return []
+            geoIds.append(row_dict[GEO_ID_COLUMN])
+    return geoIds
+
+
+def _process_new_column(var_measured_str: str,
```
[the remark this answers is not in the document]
Comment (jehangiramjad) [reply scripts/us_census/enhanced_tmcf/process_etmcf.py]: Done!

File: scripts/us_census/enhanced_tmcf/process_etmcf.py (hunk 1 above)
Comment (jehangiramjad) [reply scripts/us_census/enhanced_tmcf/process_etmcf.py:248]: Added a TODO

File: scripts/us_census/enhanced_tmcf/process_etmcf.py [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+                dcid = f"geoId/{dcid}"
+            geoId_to_dcid.update({census_geoId: dcid})
+    return geoId_to_dcid
+
+
+def _get_places_not_found(census_geoids: List[str]) -> List[str]:
+    geoId_to_dcids = _census_geoId_to_dcid(census_geoids)
+    geoIds_not_found = []
+    # Process 10 at a time.
+    num_to_process = 10
```
[the remark this answers is not in the document]
Comment (jehangiramjad) [reply scripts/us_census/enhanced_tmcf/process_etmcf.py]: done and great, I increased it to 50 for now :)

**Pull request merged**